### ISSUE TITLE
Auto-suggested '-->' doesn't overwrite when manually typing full HTML comment

### ISSUE THREAD

**Issue by ayushjaink8** [microsoft/vscode#195005]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: `1.83.0`
- OS Version: Windows 11 Version	10.0.22621 Build 22621 (Reproduced on Ubuntu 20.04.4 LTS, and Ubuntu 22.04.3 LTS as well)

Issue Description:
When manually typing out a complete HTML comment in VS Code, the auto-suggested '-->' is not getting overwritten, even though the full comment is already typed out. This creates a redundant and incorrect comment syntax.

Steps to Reproduce:
1. Open any HTML file
2. Type `<!-- Comment -->` (Do not copy-paste)

Expected Behavior:
You should see `<!-- Comment -->`.

Actual Behavior:
`<!-- Comment -->-->` is displayed.

Default Configurations:
- `quickSuggestions` for comments is enabled.
  
  ```
  "editor.quickSuggestions": {
      "comments": true
  }
  ```

Workaround:
None found yet.

Previous Version Behavior:
This issue wasn't observed in `1.82.0`.

**Comment by aeschli** [comment]
HTML uses the editor's `autoClosingPairs` feature.

**Comment by ayushjaink8** [comment]
> HTML uses the editor's `autoClosingPairs` feature.

Yes, I believe when the user is typing then the suggested `-->` could get overwritten. I am not sure but maybe it's possible to add this feature just like any other closing suggesstions get overwritten (like `"`, `}`, `]` etc.). 

More Context:
The behavior could be different in the case of tags like `</div>, </h1>`, etc. or `-->` since there is a possibility that the user might want to write "--" as a part of text (not to close the comment by typing "-->"). We can include an automatic switch like when the user starts typing "-->" then it starts get overwritten but whenever the user deviates then the the ending "-->" get auto-completed again. For example:

| S.No. | What I typed | What I see in the Editor |
| --------- | --------- | --------- |
| 1.  | < | < |
| 2. | ! | <! |
| 3. |  |  |
| 3. |  |  |
| 3. |  |  |

 "<!-- Comment"

**Comment by ayushjaink8** [comment]
> HTML uses the editor's autoClosingPairs feature.

@aeschli Yes, I believe when the user is typing then the suggested `-->` could get overwritten. I am not sure but maybe it's possible to add this feature in the text editor similar to how other **single-character** closing suggestions are handled (such as `"`, `}`, `]` etc.).

**More Context:** 
The behavior needs to be different in the case of **multi-character** closing tags (like `</div>`, `</h1>`, etc. or `-->`) since there could be instances where the user wishes to type `--` as part of the text, and not necessarily to close a comment with `-->`. To address this, we could implement an intelligent switch that activates when the user begins typing the ending `-->` sequence. In this case, the system would initiate an overwrite action but when the user deviates from typing the `-->` sequence, the editor should automatically complete the ending `-->` sequence.

Working Flow Example:
-----
**Note: `*` represents my pointer.**
| S.No. | What I Type  | What I see in the VS Code Editor |
| ------ | ------------ | ---------------------- |
| 1.      | <           | `<*`                      |
| 2.      | !           | `<!*`                     |
| 3.      | -           | `<!-*`                      |
| 4.      | -           | `<!--*-->`                     |
| 5.      | [space]           | `<!-- *-->`                    |
| 6.      |  Comment    | `<!-- Comment*-->`           |
| 7.      |  [space]    | `<!-- Comment *-->`           |

<br>

**Case 1: I exactly type "-->":**
| | | |
| ------ | ------------ | ---------------------- |
| 8.      |  -    | `<!-- Comment -*->`           |
| 9.      |  -    | `<!-- Comment --*>`           |
| 10.    |  >    | `<!-- Comment -->*`          |

**Case 2: If I type "--s" which is different from "-->" but it contains "--" which is common in both:**
| | | |
| ------ | ------------ | ---------------------- |
| 8.      |  -    | `<!-- Comment -*->`           |
| 9.      |  -    | `<!-- Comment --*>`           |
| 10.    |  s   | `<!-- Comment --s*-->`          |

**Case 3: When I hit enter after typing "--":**
| | | |
| ------ | ------------ | ---------------------- |
| 8.      |  -    | `<!-- Comment -*->`           |
| 9.      |  -    | `<!-- Comment --*>`           |
| 10.    |  [Enter]   | `<!-- Comment --`<br>`*-->`          |

--------

A similar working flow can be featured for all the auto-closing HTML tags like `</div>`, `</h1>`, `</span>` etc.

CC: @hediet
